Source/SwiftLintCore/Models/Region.swift

```diff
@@ -46,7 +46,10 @@ public struct Region: Equatable {
     ///
     /// - returns: True if the specified rule is disabled in this region.
     public func isRuleDisabled(_ rule: some Rule) -> Bool {
-        areRulesDisabled(ruleIDs: type(of: rule).description.allIdentifiers)
+        if let rule = rule as? CustomRules {
+            return areRulesDisabled(ruleIDs: rule.customRuleIdentifiers + [type(of: rule).description.identifier])
```

Comment:
Does this mean a custom rule `custom_1` will be disabled if `custom_2` is actually named in the disable command? `rule.customRuleIdentifiers` are all custom identifiers defined, aren't they?

Reply:
So this specific function will return `true` in that case, yes, but the actual identifiers will be checked afterwards, here in the construction of `identifiersWithRegions`

```
    func superfluousDisableCommandViolations(regions: [Region],
                                             superfluousDisableCommandRule: SuperfluousDisableCommandRule?,
                                             allViolations: [StyleViolation]) -> [StyleViolation] {
        guard regions.isNotEmpty, let superfluousDisableCommandRule else {
            return []
        }

        let regionsDisablingSuperfluousDisableRule = regions.filter { region in
            region.isRuleDisabled(superfluousDisableCommandRule)
        }

        let identifiersWithRegions: [(String, [Region])] = {
```

The proof of the pudding 🤞 is (slightly expanded for clarity):

```
    func testUnviolatedSpecificCustomRulesTriggersSuperfluousDisableCommand() throws {
        let customRules =         [
            "forbidden": [
                "regex": "FORBIDDEN",
            ],
            "forbidden2": [
                "regex": "FORBIDDEN2",
            ],
        ]

        let example = Example("""
                              // swiftlint:disable:next forbidden forbidden2
                              let FORBIDDEN = 1
                              """)

        let violations = try self.violations(forExample: example, customRules: customRules)
        XCTAssertEqual(violations.count, 1)
        XCTAssertTrue(violations[0].isSuperfluousDisableCommandViolation(for: "forbidden2"))
    }
```

Reply:
I'm wondering if a naming change might make this clearer, but I don't have a good candidate ...